cmd/crank/beta/validate/manager_test.go

```diff
@@ -230,6 +258,96 @@ func TestConfigurationTypeSupport(t *testing.T) {
 	}
 }
 
+func TestAddDependencies(t *testing.T) {
+	cd1 := static.NewLayer(configDep1Yaml, types.OCILayer)
+	cd2 := static.NewLayer(configDep2Yaml, types.OCILayer)
+	pd1 := static.NewLayer(providerYaml, types.OCILayer)
+	fd1 := static.NewLayer(funcYaml, types.OCILayer)
+
+	fetchMockFunc := func(image string) (*conregv1.Layer, error) {
+		switch image {
+		case "config-dep-1:v1.3.0":
+			return &cd1, nil
+		case "config-dep-2:v1.3.0":
+			return &cd2, nil
+		case "provider-dep-1:v1.3.0":
+			return &pd1, nil
+		case "function-dep-1:v1.3.0":
+			return &fd1, nil
+		default:
+			return nil, fmt.Errorf("unknown image: %s", image)
+		}
+	}
+
+	type args struct {
+		extensions []*unstructured.Unstructured
+		fetchMock  func(image string) (*conregv1.Layer, error)
+	}
+	type want struct {
+		confs int
+		deps  int
+		err   error
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"SuccessfulDependenciesAddition": {
+			// config-dep-1
+			//└─►config-dep-2
+			//config-dep-2
+			//├─►provider-dep-1
+			//└─►function-dep-1
```

Comment:
instead of listing `config-dep-2` as another entry at the root of this tree, you could just continue it's dependencies underneath its entry in the line above
```suggestion
			// config-dep-1
			//└─►config-dep-2
			//   ├─►provider-dep-1
			//   └─►function-dep-1
```